src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_postgres.py

```diff
@@ -915,21 +915,36 @@ def flexible_server_identity_update(cmd, client, resource_group_name, server_nam
     validate_resource_group(resource_group_name)
     validate_citus_cluster(cmd, resource_group_name, server_name)
 
-    identity_type = 'None'
+    server = client.get(resource_group_name, server_name)
+    identity_type = server.identity.type if (server and server.identity and server.identity.type) else 'None'
+
     if system_assigned.lower() == 'enabled':
-        identity_type = 'SystemAssigned'
+        # user wants to enable system-assigned identity
+        if identity_type == 'None':
+            # if user-assigned identity is not enabled, then enable system-assigned identity
+            identity_type = 'SystemAssigned'
+        elif identity_type == 'UserAssigned':
+            # if user-assigned identity is enabled, then enable both system-assigned and user-assigned identity
+            identity_type = 'SystemAssigned,UserAssigned'
     else:
-        server = client.get(resource_group_name, server_name)
-        identity_type = 'UserAssigned' if (server and server.identity and server.identity.type and 'UserAssigned' in server.identity.type) else 'None'
-
-    if identity_type == 'UserAssigned':
+        if server.data_encryption.type == 'AzureKeyVault':
+            # if data encryption is enabled, then system-assigned identity cannot be disabled
+            raise CLIError("Cannot disable system-assigned identity because it's used for data encryption.")
```

Comment:
@nachoalonsoportillo Data encryption is dependent on User Assigned Identity not system-assigned. If CMK enabled and has system-assigned identity, identity_type == 'SystemAssigned,UserAssigned' but would get error in this section condition.

Could you confirm this is the expected behavior you wanted change?